lilcom/lilcommodule.c

```diff
@@ -1,118 +1,193 @@
-
-  // Finding the shape of array
-  int n_samples = (int)PyArray_DIM(signalObj, 0);
-  if (n_dims != 1)
-    stride = (int)PyArray_DIM(signalObj, 1);
-
-  int **singal_stream = (int **)(PyArray_DATA(signalObj));
-
-  // Conversion to int16_t
-  int8_t *input = malloc(sizeof(int8_t) * n_samples * stride); // Allocating a linear array of considered type.
-  for (int i = 0 ; i < n_samples * stride; i++){
-    input[i] = singal_stream[i];
-  }
-
-  int16_t * output;
-  int output_stride = 1;
-  int lpc_order = 1;
+  /* Defining all variables */
+  PyObject *signal_input; // The input signal, passed as a numpy array.
+  PyObject *signal_output; // The output signal, passed as a numpy array.
+  int n_dims = 0; // Number of dimensions of the given numpy array
+  int input_stride; // The number of integers between to consecutive samples
+  int output_stride; // The number of integers between to consecutive samples in output
+  int n_samples = 0; // number of samples given in the numpy array
+  // INPUT IS DEFINED INSIDE IF CONDITIONS
+  int8_t *output; // The one dimensional vectorized output array which will be modified by the core function
+  float *input; // The one dimensional vectorized input array which will be given to the core function
+  int lpc_order = 5; // LPC Order defined in the core function (more information -> lilcom.h)
+  int conversion_exponent = 0; // Conversion Exponent defined in the core function (more information -> lilcom.h)
+
+
+  /* Reading and information - extracting for input data 
+    From the python function there are two numpy arrays and an intger (optional) LPC_order 
+    passed to this madule. Following part will parse the set of variables and store them in corresponding
+    objects.
+  */
+  static char *kwlist[] = {"X", "Y" , "lpc_order", NULL};
+  if (!PyArg_ParseTupleAndKeywords(args, keywds, "OO|i", kwlist, &signal_input, &signal_output, &lpc_order)) 
+    Py_RETURN_FALSE;
+  
+
+  printf("conversion madule = %d\n", conversion_exponent);
+  // Initializing shape related variables
+  n_dims = PyArray_NDIM(signal_input); // Getting the number of dimensions
+  n_samples = PyArray_DIM(signal_input , 0); // Getting the first dimension
+  /* In cases that numpy array is not contiguous then stride values are not equal to 1
+    The stride values given in bytes and division to the size of variables gives the 
+    strides as blocks.
+   */
+  input_stride = PyArray_STRIDE(signal_input, 0)/sizeof(float);
+  output_stride = PyArray_STRIDE(signal_output, 0)/sizeof(int8_t);
+
+
+  /* Access to the data part of the numpy array
+    PyArray_Data returns a pointer to the data stored in the memory.
+   */
+  input = PyArray_DATA(signal_input);
+  output = PyArray_DATA(signal_output);
+
+  /* Calling the core function */
+  int function_state = lilcom_compress_float(n_samples, input, input_stride, output, output_stride, lpc_order, NULL) ;
+
+  return Py_BuildValue("i",function_state);
```

Comment:
It might make sense to take account of the return codes of lilcom_compress_float because it has more than one error case.  You'll have to decide the interface; and bear in mind that we might also return false due to a dimension mismatch, so figure out how you want to assign the return codes of this wrapper function.  It might not be necessary to distinguish everything.  E.g. "1" for all "wrong arguments" calls might be OK.

Reply:
Yes I saw inside your code that you have used different error code. I think the best way to treat errors is to interpret the result to exceptions in python wrapper. Do you agree?